### PR TITLE
chore: Bump k8s modules

### PR DESCRIPTION
**Pull request by codingllama**
Update k8s modules to their latest versions.

Split from #38605.

### CODE REVIEW

**Comment by codingllama** [comment]
Branched from #38624.

**Review by rosstimothy — APPROVED** [review]

**Review by greedy52 — APPROVED** [review]

**Comment by codingllama** [comment]
Yep, I'm trying to figure out what's up.

**Comment by codingllama** [comment]
```shell
$ make enter
$ C_INCLUDE_PATH=/usr/libbpf-1.2.2/include go test -tags=bpf ./lib/bpf/... -v
# github.com/prometheus/client_golang/prometheus/testutil/promlint
/tmp/gomodcache/github.com/prometheus/client_golang@v1.18.0/prometheus/testutil/promlint/promlint.go:71:38: undefined: expfmt.FmtText
FAIL	github.com/gravitational/teleport/lib/bpf [build failed]
FAIL
```

File: go.mod [hunk 1]
```diff
@@ -149,7 +149,7 @@ require (
 	github.com/pquerna/otp v1.4.0
 	github.com/prometheus/client_golang v1.18.0
 	github.com/prometheus/client_model v0.5.0
-	github.com/prometheus/common v0.46.0
+	github.com/prometheus/common v0.45.0
```
Comment (codingllama) [inline go.mod:152]: This matches prometheus/client_golang v1.18.0: https://github.com/prometheus/client_golang/blob/v1.18.0/go.mod#L10-L12

**Pull request merged**